cns/restserver/ipam.go

```diff
@@ -112,6 +112,28 @@ func (service *HTTPRestService) MarkIPsAsPending(numberToMark int) (map[string]c
 	return nil, fmt.Errorf("Failed to mark %d IP's as pending, only marked %d IP's", numberToMark, len(pendingReleaseIPs))
 }
 
+// UpdatePendingProgrammingIPs will update pending programming IPs to available if
+// NMAgent side's programmed NC version keep up with NC version attached with secondary IP.
+func (service *HTTPRestService) UpdatePendingProgrammingIPs(nmagentNCVersion string) error {
+	service.Lock()
+	defer service.Unlock()
+	//for uuid, ipConfigurationStatus := range service.PodIPConfigState {
+	for _, containerstatus := range service.state.ContainerStatus {
+		for uuid, secondaryIPConfigs := range containerstatus.CreateNetworkContainerRequest.SecondaryIPConfigs {
+			ipConfigStatus, exist := service.PodIPConfigState[uuid]
+			if exist {
+				// TODO change cns.Available to cns.PendingProgrammiong when #690 merged.
+				// TODO change ipConfigStatus.IPAddress to ipConfigStatus.NCVersion when #697 merged.
+				if ipConfigStatus.State == cns.Available && secondaryIPConfigs.IPAddress <= nmagentNCVersion {
```

Comment:
Not sure if I understood this change here. We are moving from Pending -> Available, no? Also you are comparing ipaddress to a version. This a bug. Can you add a UT for this please

Reply:
The TODO sentence ahead of this change mentioned it.
// TODO change cns.Available to cns.PendingProgrammiong when #690 merged.
// TODO change ipConfigStatus.IPAddress to ipConfigStatus.NCVersion when #697 merged.